Sage/carbon
# Fe 5708 select outside click in dialog

### Proposed behaviour



When a select component is inside a dialog and the select list is closed, it should not cause focus to be automatically placed in the select textbox. This causes a particular problem when `openOnFocus` is set on the select, in which case clicking outside fails to close the select (or rather, it does close it but then immediately refocuses the select and therefore reopens the list - see codesandbox in linked issue for a demonstration)

I have also refactored the `useModalManager` hook, which previously took 4 arguments and as a result of this fix would need 5 - it is difficult to remember what all these arguments are and in which order they go, making it feel unclean even with autocompletion to help. It now takes a single object as an argument, making it easy to not pass unneeded arguments and for readers to see what is happening when the hook is called.

### Current behaviour

Selects are always refocused when the list is closed when inside a dialog.



### Checklist




#### QA


### Additional context



### Testing instructions



Test the sandbox from the issue and ensure the issue is fixed.

Run regression testing on select and on all modal components.
